signer/src/blocklist_client.rs

```diff
@@ -28,14 +27,17 @@ pub struct BlocklistClient {
 }
 
 impl BlocklistChecker for BlocklistClient {
-    fn can_accept<'a>(&'a self, address: &'a str) -> Pin<Box<dyn Future<Output = Result<bool, ClientError<CheckAddressError>>> + Send + 'a>> {
+    fn can_accept(
+        &self,
+        address: &str,
+    ) -> impl Future<Output = Result<bool, ClientError<CheckAddressError>>> + Send {
```

Comment:
One interesting about `async` in this case is that you can simplify it even further. The compiler checks whether the function actually implements the required traits specified in the trait, so you can just write it like a "normal" `async` function. So I think something like this will suffice.
```rust
    async fn can_accept(&self, address: &str) -> Result<bool, ClientError<CheckAddressError>> {
       let config = self.config.clone();

        // Call the generated function from blocklist-api
        let resp: BlocklistStatus = check_address(&config, address).await?;
        Ok(resp.accept)
    }
```

Reply:
Oh interesting, updated. Thanks!